// Generated by Create API
// https://github.com/CreateAPI/CreateAPI

import Foundation

/// Occurs when an [error](/docs/guides/error-codes#api-errors) occurs. This can happen due to an internal server error or a timeout.
public struct ErrorEvent: Codable {
    public var event: Event
    public var data: Error

    public enum Event: String, Codable, CaseIterable {
        case error
    }

    public init(event: Event, data: Error) {
        self.event = event
        self.data = data
    }

    public init(from decoder: Decoder) throws {
        let values = try decoder.container(keyedBy: StringCodingKey.self)
        self.event = try values.decode(Event.self, forKey: "event")
        self.data = try values.decode(Error.self, forKey: "data")
    }

    public func encode(to encoder: Encoder) throws {
        var values = encoder.container(keyedBy: StringCodingKey.self)
        try values.encode(event, forKey: "event")
        try values.encode(data, forKey: "data")
    }
}
